### ISSUE TITLE
OnFlushCompleted() is not called if DB option `atomic_flush` is set true

### ISSUE THREAD

**Issue by zhangyifan27** [facebook/rocksdb#8495]
### Expected behavior
OnFlushCompleted() should be called once the flush is complete.

### Actual behavior
OnFlushCompleted() was not called when`atomic_flush` is set true.

### Steps to reproduce the behavior
Reproduce this by setting `atomic_flush` true in `EventListenerTest.MultiCF` ([c53d1e](https://github.com/zhangyifan27/rocksdb/commit/c53d1ea5d9ef9e58f04995cf860bc92c8b011390)) and the test failed.

```
[ RUN      ] EventListenerTest.MultiCF
db/listener_test.cc:378: Failure
Expected equality of these values:
  listener->flushed_dbs_.size()
    Which is: 0
  i
    Which is: 1
[  FAILED  ] EventListenerTest.MultiCF (1555 ms)
```

**Comment by riversand963** [comment]
In `MemTableList::TryInstallMemTableFlushResults()`, we populate an out arg `committed_flush_jobs_info`. We should do the same for `MemTableList::InstallMemtableAtomicFlushResults()`. Otherwise, the flush job's `committed_flush_jobs_info_` will always remain empty if `atomic_flush=true`. Consequently, `NotifyOnFlushCompleted()` will skip this cf.

**Comment by riversand963** [comment]
should be fixed by #8585
